src/libNetwork/P2PComm.cpp

```diff
@@ -275,20 +336,26 @@ void SendJob::SendMessageCore(const Peer& peer, const bytes message,
                               unsigned char startbyte, const bytes hash) {
   uint32_t retry_counter = 0;
   while (!SendMessageSocketCore(peer, message, startbyte, hash)) {
-    retry_counter++;
-    LOG_GENERAL(WARNING, "Socket connect failed " << retry_counter << "/"
-                                                  << MAXRETRYCONN
-                                                  << ". IP address: " << peer);
-
-    if (P2PComm::IsHostHavingNetworkIssue()) {
+    // comment this since we already check this in SendMessageSocketCore() and
+    // also add to blacklist
+    /*if (P2PComm::IsHostHavingNetworkIssue()) {
       LOG_GENERAL(WARNING, "[blacklist] Encountered "
                                << errno << " (" << std::strerror(errno)
                                << "). Adding " << peer.GetPrintableIPAddress()
                                << " to blacklist");
       Blacklist::GetInstance().Add(peer.m_ipAddress);
       return;
+    }*/
+
+    if (Blacklist::GetInstance().Exist(peer.m_ipAddress)) {
+      return;
     }
 
+    retry_counter++;
+    LOG_GENERAL(WARNING, "Socket connect failed " << retry_counter << "/"
```

Comment:
shall we move the logging to before `retry_counter++;`. I think it is more accurate that way.

Reply:
yeah.